continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -72,19 +72,19 @@ object DefDefTransforms extends TreesChecks:
         report.logWith("state machine and new defdef:")(transformedTree)
 
     tree match
-      case HasSuspensionNotInReturnedValue(_) =>
+      case _ if HasSuspensionNotInReturnedValue(tree) =>
         transformSuspensionsSuspendingStateMachine(fetchSuspensions, false)
-      case CallsSuspendContinuation(_) =>
+      case _ if CallsSuspendContinuation(tree) =>
         fetchSuspensions match
           case suspensionPoint :: Nil if !suspensionPoint.isInstanceOf[tpd.ValDef] =>
             transformSuspendOneContinuationResume(tree, suspensionPoint)
           case suspensionPoints =>
             transformSuspensionsSuspendingStateMachine(suspensionPoints, true)
-      case BodyHasSuspensionPoint(_) =>
+      case vd: tpd.DefDef if BodyHasSuspensionPoint(vd) =>
         // any suspension that still needs a transformation
         tree match
           case t: tpd.DefDef => cpy.DefDef(t)()
-          case t: tpd.ValDef => cpy.ValDef(t)()
+      // case t: tpd.ValDef => cpy.ValDef(t)()
```

Comment:
this is necessary because tree can be def or val https://github.com/47deg/TBD/blob/23c21f31779182f74ae922ec0cae1a872a92b20f/continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala#L55.

we did some work on `val` but we de-prioritised for now